pages/docs/tutorials/native/basic-kotlin-native-app.md

```diff
@@ -2,90 +2,275 @@
+authors: 
+  - Hadi Hariri
+  - Eugene Petrenko 
+date: 2019-04-05
 ---
 
 
-In this tutorial we'll see how to
+In this tutorial, we'll look at how to
 
 * [Obtain the Kotlin/Native compiler](#obtaining-the-compiler)
 * [Write the application](#creating-hello-kotlin)
-* [Compile and examine output](#compiling-and-examining-output)
+* [Create build files](#creating-a-kotlinnative-gradle-project)
+* [Set up an IDE](#open-in-ide)
 * [Run the application](#running-the-application)
 
 
-## Obtaining the compiler
+## Obtaining the Compiler
 
-Kotlin/Native is available for macOS, Linux and Windows. Depending on the operating system we're working on, we'll need to download
-the correct compiler. While cross-platform compilation is possible (i.e. using one platform to compile for another), in this first tutorial
-we're going to compile for the same operating system we're running on. In our case this will be macOS. 
+Kotlin/Native is available for macOS, Linux, and Windows. While cross-platform compilation is possible
+(i.e., using one platform to compile for another), in this first tutorial
+we are only compiling for the operating system we're running on. 
 
-We can obtain the latest version of the compiler from the [GitHub releases page](https://github.com/JetBrains/kotlin-native/releases).
+The best way to use the Kotlin/Native compiler is with a build system.
+It helps by downloading and caching the Kotlin/Native compiler binaries and libraries with
+transitive dependencies, and running the compiler and tests.
+The compilation results are cached too.
+A build system can also be used by an IDE to understand the project layout.
 
-Once downloaded, we can uncompress it in any folder, e.g. ~/kotlin-native. For convenience it's useful to add the `bin` folder to the system path so that we can invoke the 
-compiler from any location. (If we uncompressed to ~/kotlin-native, this would be ~/kotlin-native/bin) 
+Kotlin/Native uses the [Gradle](https://gradle.org) build system through the
+[kotlin-multiplatform](/docs/reference/building-mpp-with-gradle.html) plugin.
+We'll look at how to configure a Gradle build below.
+For some corner cases, a Kotlin/Native compiler can still be obtained manually (not recommended)
+from the [Kotlin releases page on GitHub](https://github.com/JetBrains/kotlin/releases).
+In the tutorial, we are focussing on using the Gradle builds. 
 
-While the output by the compiler does not have any dependencies, the compiler itself does require Java 8, which should be on the system. If we're using macOS, we also need the macOS SDK which can be installed by installing Xcode. 
+While the output of the compiler does not have any dependencies or virtual machine requirements,
+the compiler itself and the Gradle build system require a Java 1.8 or 11 runtime. Check out the 
+[https://jdk.java.net/11](https://jdk.java.net/11/) or [https://adoptopenjdk.net/](https://adoptopenjdk.net/)
```

Comment:
I don't find this totally ok to recommend downloading binaries from an unofficial third-party site.
Why one may need more than jdk.java.net?